test/template_method_linux_test.cpp

```diff
@@ -0,0 +1,88 @@
+// For only Linux testing
+
+#include <boost/dll/runtime_symbol_info.hpp>
+#include <boost/dll/smart_library.hpp>
+#include <boost/dll/import_mangled.hpp>
+#include <boost/dll/import_class.hpp>
+#include <iostream>
+#include <memory>
+
```

Comment:
I'll register the test in build files. Jamfiles are hard and creepy, you do not have to deal with them